grails-app/controllers/com/secured/auth/UserController.groovy

```diff
@@ -36,37 +37,45 @@ class UserController  {
         {
             cmd.errors.rejectValue("password",
                                "user.password.doesntmatch")
+            cmd.password = ""
             flash.model = cmd
             redirect(action:"register")
             return
         }
-        if(cmd.validate())
+        if(!cmd.validate())
         {
-            User usr = User.createUser(cmd)
-            def userRole = Role.findOrSaveWhere(authority: "ROLE_USER")
-            if(userValidator.isUsernameValid(usr)  && userValidator.isPasswordValid(usr))
-            {
-                usr.save()
-                userInitializer.assignRole(usr,userRole,true)
-                String message = "Click the link below to verify your email\n " +
-                        "localhost:8080/verify?token=${usr.mainToken}"
-                new Mail()
-                    .from("your email")
-                    .to(usr.email)
-                    .withSubject("Confirm your email")
-                    .withMessage(message)
-                    .useSendingStrategy(SendViaGmail.usingAccount("your username",
-                                "your password"))
-                    .onErrors(RejectEmail.withMessage("Something went wrong"))
-                    .send()
-                springSecurityService.reauthenticate(usr.username,usr.password)
-                redirect controller: 'main',action:'confirm'
-                return
-            }
+            flash.model = cmd
+            redirect(action:"register")
+            return
         }
-        flash.model = cmd
-        redirect(action:"register")
-
+        if(!userValidator.isPasswordValid(cmd))
+        {
+            flash.model = cmd
+            redirect(action:"register")
```

Comment:
do not send model by flash 
redirect(action:"register", model :[cmd: cmd])

Reply:
The same for the others

Reply:
redirect methodin map poxanceluc inqy mapum model parametri chi spasum ughaki, dra hamar tenc ban chem arel, lavaguyn depqum karam senc grem redirect(action: "register", params: params) u register actionum grem UserCommand cmd = UserCommand.create(params) kam cmd-n tam vorpes parameter